src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderChapter.jsx

```diff
@@ -0,0 +1,373 @@
+  );
+}
+
+function throwIncorrectItemPath() {
+  throw new Error(
+    'arrayBuilderChapter item pages must include a `path` property that ends with `/:index`',
+  );
+}
+
+function verifyRequiredOptions(pageType, requiredOpts, objectToCheck) {
+  requiredOpts.forEach(option => {
+    if (objectToCheck[option] === undefined) {
+      if (pageType === 'chapter') {
+        throwErrorChapter(option);
+      } else {
+        throwErrorPage(pageType, option);
+      }
+    }
+  });
+}
+
+function determineYesNoField(uiSchema) {
+  let yesNoKey;
+  for (const key of Object.keys(uiSchema)) {
+    if (uiSchema[key]['ui:webComponentField'] === YesNoField) {
+      yesNoKey = key;
+    }
+  }
+  return yesNoKey;
+}
+
+/**
+ * arrayBuilderChapter
+ *
+ * Example:
+ * ```
+ * chapterName: arrayBuilderChapter(pageBuilder => {
+ *   title: 'Employment history',
+ *   options: {
+ *     arrayPath: 'employers',
+ *     noun: 'employer',
+ *     nextChapterPath: '/next-chapter',
+ *   }
+ *   summaryPage: pageBuilder.summaryPage({
+ *     title: 'Employment history',
+ *     path: '/summary',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ *   itemFirstPage: pageBuilder.itemFirstPage({
+ *     title: 'Name and address of employer or unit',
+ *     path: '/employer',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ *   itemLastPage: pageBuilder.itemLastPage({
+ *     title: 'Name and address of employer or unit',
+ *     path: '/employer',
+ *     uiSchema: ...,
+ *     schema: ...
+ *   }),
+ * }
+ * ```
+ *
+ * @param {(pageBuilder: ArrayBuilderPages) => ArrayBuilderConfig} configFunction
+ * @returns {FormConfigChapter}
+ */
+export default function arrayBuilderChapter(configFunction) {
+  let itemFirstPagePath;
+  let summaryPath;
+  let hasItemsKey;
+
+  if (!configFunction || typeof configFunction !== 'function') {
+    throwErrorNoConfigFunction();
+  }
+
+  /**
+   * @type {{
+   *   [key: string]: (config: FormConfigPage) => FormConfigPage,
+   * }}
+   */
+  const pageBuilder = {
+    summaryPage: pageConfig => {
+      summaryPath = pageConfig.path;
+      try {
+        hasItemsKey = determineYesNoField(pageConfig.uiSchema);
+      } catch (e) {
+        throwMissingYesNoField();
+      }
+      if (!hasItemsKey) {
+        throwMissingYesNoField();
+      }
+    },
+    itemFirstPage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+      itemFirstPagePath = getBasePath(pageConfig.path);
+    },
+    itemMiddlePage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+    itemLastPage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
+        throwIncorrectItemPath();
+      }
+    },
+    itemSinglePage: pageConfig => {
+      if (!pageConfig?.path.endsWith('/:index')) {
```

Comment:
There are several forms that include an `:index` in the middle of the path, e.g. `applicant-information/:index/medicare-status`. Not sure if this needs to be accounted for in this pattern.

Reply:
Also, does this mean that there are a maximum number of non-summary pages of 3?

Reply:
- oh interesting about the :index in the middle. I'll test that. Thanks for raising. 
- You can have unlimited pages